frame/election-provider-multi-phase/test-election-provider/src/mock.rs

```diff
@@ -0,0 +1,494 @@
+	type Reward = (); // rewards are minted from the void
+	type SessionsPerEra = SessionsPerEra;
+	type BondingDuration = BondingDuration;
+	type SlashDeferDuration = SlashDeferDuration;
+	type SlashCancelOrigin = EnsureRoot<AccountId>; // root can cancel slashes
+	type SessionInterface = Self;
+	type EraPayout = ();
+	type NextNewSession = Session;
+	type MaxNominatorRewardedPerValidator = MaxNominatorRewardedPerValidator;
+	type OffendingValidatorsThreshold = OffendingValidatorsThreshold;
+	type ElectionProvider = ElectionProviderMultiPhase;
+	type GenesisElectionProvider = onchain::OnChainExecution<OnChainSeqPhragmen>;
+	type VoterList = BagsList;
+	type TargetList = pallet_staking::UseValidatorsMap<Self>;
+	type MaxUnlockingChunks = ConstU32<32>;
+	type HistoryDepth = HistoryDepth;
+	type OnStakerSlash = ();
+	type WeightInfo = pallet_staking::weights::SubstrateWeight<Runtime>;
+	type BenchmarkingConfig = StakingBenchmarkingConfig;
+}
+
+impl<LocalCall> frame_system::offchain::SendTransactionTypes<LocalCall> for Runtime
+where
+	RuntimeCall: From<LocalCall>,
+{
+	type OverarchingCall = RuntimeCall;
+	type Extrinsic = Extrinsic;
+}
+
+pub struct OnChainSeqPhragmen;
+
+parameter_types! {
+	pub VotersBound: u32 = 600;
+	pub TargetsBound: u32 = 400;
+}
+
+impl onchain::Config for OnChainSeqPhragmen {
+	type System = Runtime;
+	type Solver = SequentialPhragmen<
+		AccountId,
+		pallet_election_provider_multi_phase::SolutionAccuracyOf<Runtime>,
+	>;
+	type DataProvider = Staking;
+	type WeightInfo = ();
+	type MaxWinners = MaxWinners;
+	type VotersBound = VotersBound;
+	type TargetsBound = TargetsBound;
+}
+
+pub struct StakingBenchmarkingConfig;
+impl pallet_staking::BenchmarkingConfig for StakingBenchmarkingConfig {
+	type MaxNominators = traits::ConstU32<1000>;
+	type MaxValidators = traits::ConstU32<1000>;
+}
+
+pub struct ElectionProviderBenchmarkConfig;
+
+impl pallet_election_provider_multi_phase::BenchmarkingConfig for ElectionProviderBenchmarkConfig {
+	const VOTERS: [u32; 2] = [1000, 2000];
+	const TARGETS: [u32; 2] = [500, 1000];
+	const ACTIVE_VOTERS: [u32; 2] = [500, 800];
+	const DESIRED_TARGETS: [u32; 2] = [200, 400];
+	const SNAPSHOT_MAXIMUM_VOTERS: u32 = 1000;
+	const MINER_MAXIMUM_VOTERS: u32 = 1000;
+	const MAXIMUM_TARGETS: u32 = 300;
+}
+
+pub struct OtherSessionHandler;
+impl traits::OneSessionHandler<AccountId> for OtherSessionHandler {
+	type Key = testing::UintAuthorityId;
+
+	fn on_genesis_session<'a, I: 'a>(_: I)
+	where
+		I: Iterator<Item = (&'a AccountId, Self::Key)>,
+		AccountId: 'a,
+	{
+	}
+
+	fn on_new_session<'a, I: 'a>(_: bool, _: I, _: I)
+	where
+		I: Iterator<Item = (&'a AccountId, Self::Key)>,
+		AccountId: 'a,
+	{
+	}
+
+	fn on_disabled(_validator_index: u32) {}
+}
+
+impl sp_runtime::BoundToRuntimeAppPublic for OtherSessionHandler {
+	type Public = testing::UintAuthorityId;
+}
+
+#[derive(Default)]
+pub struct ExtBuilder;
+
+pub mod agents {
+	use super::AccountId;
+
+	pub const ACCOUNT_0: AccountId = 0;
```

Comment:
I advise you to revise this, but let's see how it will pan out.

Reply:
as in, there could be a rule for accounts, and a code can generate those rules. for example: 

All accounts from 10 to 20 (or 10 + `v`) are validators.
For each validator `x`, the accounts `[2 * x, .. 2 * x + 5]` would be their nominators, assuming we have 5 nominators their validator etc.

Just an idea about how you can take this forward. I've learned that managing accounts in tests can become a bit nontrivial and a PITA to remember in the future if not done well.

Reply:
I was trying to see if there was a sane way to work with account in the tests, but I think the experiment didn't work as I was expecting. Reverted to as its done across all pallet tests.